File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Weakness.java

package com.watabou.pixeldungeon.actors.buffs;

import com.watabou.pixeldungeon.actors.Char;
import com.watabou.pixeldungeon.items.rings.RingOfElements.Resistance;
import com.watabou.pixeldungeon.ui.BuffIndicator;

import org.jetbrains.annotations.NotNull;

public class Weakness extends FlavourBuff {

	private static final float DURATION = 40f;

	@Override
	public int icon() {
		return BuffIndicator.WEAKNESS;
	}
	
	@Override
	public boolean attachTo(@NotNull Char target ) {
		if (super.attachTo( target )) {
			target.getBelongings().discharge();
			return true;
		} else {
			return false;
		}
	}

	public static float duration( Char ch ) {
		Resistance r = ch.buff( Resistance.class );
		return r != null ? r.durationFactor() * DURATION : DURATION;
	}
}

File: scripts/buffs/Weakness.lua
---
--- Weakness: discharges equipped items on attach (was actors/buffs/Weakness.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

return buff.init{
    desc  = function ()
        return {
            icon          = 14, -- BuffIndicator.WEAKNESS
            name          = "WeaknessBuff_Name",
            info          = "WeaknessBuff_Info",
        }
    end,

    attached = function(self, buff)
        buff.target:getBelongings():discharge()
    end
}
